helix-core/src/main/java/org/apache/helix/model/Message.java

```diff
@@ -173,6 +181,14 @@ public void setCreateTimeStamp(long timestamp) {
     _record.setLongField(Attributes.CREATE_TIMESTAMP.toString(), timestamp);
   }
 
+  /**
+   * Set the time that the message was expected to be completed
+   * @param timestamp a UNIX epoch timestamp
+   */
+  public void setCompletionDueTimeStamp(long timestamp) {
+    _record.setLongField(Attributes.COMPLETION_DUE_TIMESTAMP.name(), timestamp);
```

Comment:
Do we need any validation here? For example, it timestamp already < currentmilisecond.

Reply:
Is that a problem, for example, I may set an time before current time to invalid the message immediately?